src/mac.m

```diff
@@ -1205,48 +1203,27 @@ - (void)windowDidResignKey:(NSNotification*)notification
 }
 #endif
 
-static void
-dispatchClientEvent(PuglWorld* world, NSEvent* ev)
-{
-  NSWindow* win = [ev window];
-  NSPoint   loc = [ev locationInWindow];
-  for (size_t i = 0; i < world->numViews; ++i) {
-    PuglView*        view    = world->views[i];
-    PuglWrapperView* wrapper = view->impl->wrapperView;
-    if ([wrapper window] == win && NSPointInRect(loc, [wrapper frame])) {
-      const PuglClientEvent event = {
-        PUGL_CLIENT, 0, (uintptr_t)[ev data1], (uintptr_t)[ev data2]};
-
-      PuglEvent clientEvent;
-      clientEvent.client = event;
-      puglDispatchEvent(view, &clientEvent);
-    }
-  }
-}
-
 PuglStatus
 puglUpdate(PuglWorld* world, const double timeout)
 {
   @autoreleasepool {
-    NSDate* date =
-      ((timeout < 0) ? [NSDate distantFuture]
-                     : [NSDate dateWithTimeIntervalSinceNow:timeout]);
-
-    for (NSEvent* ev = NULL;
-         (ev = [world->impl->app nextEventMatchingMask:NSAnyEventMask
-                                             untilDate:date
-                                                inMode:NSDefaultRunLoopMode
-                                               dequeue:YES]);) {
-      if ([ev type] == NSApplicationDefined && [ev subtype] == (NSEventSubtype)PUGL_CLIENT) {
-        dispatchClientEvent(world, ev);
-      }
-
-      [world->impl->app sendEvent:ev];
-
-      if (timeout < 0) {
-        // Now that we've waited and got an event, set the date to now to avoid
-        // looping forever
-        date = [NSDate date];
+    if (world->impl->autoreleasePool != nil) {
```

Comment:
This check looks a little weird, might be worth mentioning that `autoreleasePool` is only non-null for `PUGL_PROGRAM` types.

Reply:
I mean mention as a code comment.

Reply:
It might be nicer to introduce a `PuglWorldType puglWorldGetType (PuglWorld const*)` which could be used in this location. That would require implementation for other platforms too, though.

Reply:
The guts of the world/everything are available anywhere in the implementation, so the accessor part isn't really necessary (though it doesn't hurt either).